### PR TITLE
feat(stdlib): Implement ifArrayJoin interface for roArray

### LINKED ISSUE #125: implement ifArrayJoin interface for roArray
ifArrayJoin includes the join() function on roArray, which is an important and often used function in many codebases.  

https://sdkdocs.roku.com/display/sdkdoc/ifArrayJoin

### PR DESCRIPTION
**Pull request by lvcabral**
fixes #125

### CODE REVIEW

File: src/brsTypes/components/RoArray.ts [hunk 1]
```diff
@@ -180,4 +181,21 @@ export class RoArray extends BrsComponent implements BrsValue, BrsIterable {
...
+            args: [new StdlibArgument("separator", ValueKind.String)],
+            returns: ValueKind.String,
+        },
+        impl: (interpreter: Interpreter, separator: BrsString) => {
+            if (
+                this.elements.some(function(element) {
+                    return !(element instanceof BrsString);
+                })
+            ) {
+                throw new Error("roArray.Join: Array contains non-string value(s).");
```
Comment (sjbarag) [inline src/brsTypes/components/RoArray.ts]: Watch out — it looks like the reference BrightScript implementation returns an empty string here.  Executing this file:

```brightscript
sub main()
    a = [1, 2, 3].join(",")
    print a
    print type(a)
end sub
```

results in this output:

```
BRIGHTSCRIPT: ERROR: roArray.Join: non-string value at index 0: pkg:/source/main.brs(3)

String
```

Please update this implementation and add a unit test to ensure that works 😃

**Comment by lvcabral** [comment]
I will take a look at this case later today. I'm currently working on ifArraySort interface too

**Comment by lvcabral** [comment]
Sorry it mixed up ifArrayJoin with ifArraySort, I could not figure out how to split these two pull requests.

**Comment by lvcabral** [comment]
Closing this PR as it was already integrated.

**Pull request closed without merging**